app/lib/pre_assembly/digital_object.rb

```diff
@@ -81,10 +82,10 @@ def pre_assemble
         end
       end
 
-      if ObjectFileValidator.new(object: self, batch:).object_has_hierarchy? && content_structure != 'file'
+      unless object_files_valid?
```

Comment:
It doesn't seem quite idiomatic to set @object_validation_message as a by-product of calling object_files_valid. How about returning the message or nil?

```
if (object_validation_message = object_files_valid?)
```

Reply:
@justinlittman yeah, I went back and forth on that pattern and this one while trying to limit the complexity increase to `pre_assemble` -- I'll rework to be a bit more idiomatic.